test/box-luatest/gh_6452_mvcc_crash_in_prepare_test.lua

```diff
@@ -0,0 +1,84 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{
+        alias   = 'default',
+        box_cfg = {memtx_use_mvcc_engine = true}
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_mvcc_crash_in_prepare_case1 = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local txn_proxy = require("test.box.lua.txn_proxy")
+
+        box.cfg{memtx_use_mvcc_engine = true}
+        local s = box.schema.space.create("s", {engine="memtx"})
+        s:create_index("pk", {type="tree"})
+        s:create_index("sk", {parts={2}, type="hash"})
+
+        local tx1 = txn_proxy.new()
+        local tx2 = txn_proxy.new()
+        local tx3 = txn_proxy.new()
+        local tx4 = txn_proxy.new()
+
+        tx1:begin()
+        tx2:begin()
+        tx3:begin()
+        tx4:begin()
+
+        tx1("box.space.s:insert{1, 'A'}")
+        tx1:commit()
+        tx2("box.space.s:replace{2, 'B'}")
+        tx3("box.space.s:replace{1, 'B'}")
+        tx4("box.space.s:replace{3, 'B'}")
+
```

Comment:
I think transaction tests should illustrate conflicts as clearly as possible: consider replacing your test with a more simple one, for instance
```lua
tx1("box.space.s:insert{1, 'A'}")
tx2("box.space.s:insert{2, 'B'}")
tx3("box.space.s:replace{1, 'B'}")
tx1:commit()
tx2:commit()
```

Reply:
Great test, thanks.